ShapesProject/Project1/Main.cpp

```diff
@@ -0,0 +1,171 @@
+#include <iostream>
+#include "FShape.h"
+#include "FCircle.h"
+#include "FSquare.h"
+#include "TStaticArray.h"
+#include "TDynamicArray.h"
+
+#include "ValidateInput.h"
+
+#define Print(Text) std::cout << Text << std::endl;
+
+//------------------------------------------------------------------------------------------
+
+FCircle* GetCircle()
+{
+	float Radius = GetValidateInput("Enter the radius: ");
+	FCircle* NewCircle = new FCircle(Radius);
+
+	return NewCircle;
+}
+
+//------------------------------------------------------------------------------------------
+
+FSquare* GetSquare()
+{
+	float Size = GetValidateInput("Enter the Size of the edges: ");
+	FSquare* NewSquare = new FSquare(Size);
+	return NewSquare;
+}
+
+//------------------------------------------------------------------------------------------
+
+template<typename T>
+void PrintArray(const T& InArray, int InSize)
+{
+	system("cls");
+	Print("--------------------------Array Content--------------------------");
+	for (int i = 0; i < InSize; i++)
+	{
+		float LoopArea = InArray[i]->GetArea();
+		float LoopPerimeter = InArray[i]->GetPerimeter();
+
+		Print("Shape " << i << " area: " << LoopArea << " perimeter: " << LoopPerimeter);
+	}
+
+	std::cin.ignore();
+	std::cin.get();
+}
+
+//------------------------------------------------------------------------------------------
+
+FShape* AskForShapes()
+{
+	system("cls");
+	int ShapeChoice = GetValidateInput(1, 2, "\n 1 - Cirle \n 2 - Square \n Shape: ");
+	if (ShapeChoice == 1)
+	{
+		return GetCircle();
+	}
+	else if (ShapeChoice == 2)
+	{
+		return GetSquare();
+	}
+}
+
+//------------------------------------------------------------------------------------------
+
+void AskForShapesBuiltIn()
+{
+	const int ShapesNumber = GetValidateInput("How many figures do you want to create?");;
+
+	FShape** BuiltInArray = new FShape *[ShapesNumber];
+
+	Print("Specify what figures you want.");
+
+	for (int i = 0; i < ShapesNumber; i++) 
+	{
+		BuiltInArray[i] = AskForShapes();
+	}
+
+	Print("----------------Saved Shapes----------------");
+
+	PrintArray(BuiltInArray, ShapesNumber);
+
+	for (int i = 0; i < ShapesNumber; i++)
+	{
+		delete BuiltInArray[i];
```

Comment:
This is a better use for the delete inside the array, you now know these are in fact stored in the heap memory